### PR TITLE
Remaining Cabal-Install package errors

### PR DESCRIPTION
**Pull request by SuganyaAK**
Remaining Cabal-Install Package Exceptions followed by PR #9276 and #9219.
---

**Template Α: This PR modifies `cabal` behaviour**

Include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.
* [ ] [Manual QA notes](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#qa-notes) have been included.

Bonus points for added automated tests!

---

**Template Β: This PR does not modify `cabal` behaviour (documentation, tests, refactoring, etc.)**

Include the following checklist in your PR:

* [ ✅] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).

### CODE REVIEW

**Review by geekosaur — APPROVED** [review]

File: cabal-install/src/Distribution/Client/Errors.hs [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# OPTIONS_GHC -fno-warn-incomplete-uni-patterns #-}
```
Comment (ffaf1) [inline cabal-install/src/Distribution/Client/Errors.hs:6]: This looks a bit suspect, why is it needed?

File: cabal-install/src/Distribution/Client/Errors.hs [hunk 2]
```diff
@@ -1,6 +1,9 @@
 {-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE NamedFieldPuns #-}
```
Comment (ffaf1) [inline cabal-install/src/Distribution/Client/Errors.hs:5]: I am not a fan of Field Puns, where are you using them?

**Comment by ffaf1** [comment]
Woops sorry, did not see this was a draft. Feel free to ignore my comments.

File: cabal-install/src/Distribution/Client/Errors.hs (hunk 1 above)
Comment (SuganyaAK) [reply cabal-install/src/Distribution/Client/Errors.hs:6]: This is part of existing code (TODO in module `TargetSelector.hs`). I have moved the existing logic to `Errors.hs`, I did not want to make changes to existing code / logic, so I left it unchanged.

**Review by gbaz — APPROVED** [review]

**Pull request merged**